core/domain/feedback_services.py

```diff
@@ -139,81 +139,142 @@ def create_message(
+                if thread.entity_type == feconf.ENTITY_TYPE_EXPLORATION:
+                    (
+                        event_services
+                        .FeedbackThreadStatusChangedEventHandler
+                        .record(
+                            thread.entity_id, thread.status, updated_status)
+                    )
+        if updated_subject:
+            message.updated_subject = updated_subject
+    feedback_models.GeneralFeedbackMessageModel.put_multi(messages)
+
+    # Update the message data cache of the threads.
+    for thread in thread_models:
         thread.message_count += 1
-    else:
-        thread.message_count = (
-            feedback_models.GeneralFeedbackMessageModel.get_message_count(
-                thread_id))
+        if text:
+            thread.last_nonempty_message_text = text
+            thread.last_nonempty_message_author_id = author_id
 
     # We do a put() even if the status and subject are not updated, so that the
-    # last_updated time of the thread reflects the last time a message was added
-    # to it.
-    old_status = thread.status
-    if message_id != 0 and (updated_status or updated_subject):
-        if updated_status and updated_status != thread.status:
-            thread.status = updated_status
-        if updated_subject and updated_subject != thread.subject:
-            thread.subject = updated_subject
-    new_status = thread.status
-    thread.put()
-
-    # We do a put on the suggestion linked (if it exists) to the thread, so that
-    # the last_updated time changes to show that there is activity in the
-    # thread.
-    if thread.has_suggestion:
-        suggestion_id = thread_id
-        suggestion = (
-            suggestion_models.GeneralSuggestionModel.get_by_id(suggestion_id))
+    # last_updated time of the threads reflects the last time a message was
+    # added to it.
+    old_statuses = [thread.status for thread in thread_models]
+    new_statuses = old_statuses
+    if updated_status or updated_subject:
+        new_statuses = []
+        for index, thread in enumerate(thread_models):
+            # Can't be the first thread.
+            if message_ids[index] != 0:
+                if updated_status and (updated_status != thread.status):
+                    thread.status = updated_status
+                if updated_subject and (updated_subject != thread.subject):
+                    thread.subject = updated_subject
+            new_statuses.append(thread.status)
+    feedback_models.GeneralFeedbackThreadModel.put_multi(thread_models)
+
+    # For each thread, we do a put on the suggestion linked (if it exists) to
+    # the thread, so that the last_updated time changes to show that there is
+    # activity in the thread.
+    thread_ids_that_have_linked_suggestions = []
+    for thread in thread_models:
+        if thread.has_suggestion:
+            thread_ids_that_have_linked_suggestions.append(thread.id)
+    suggestions = (
```

Comment:
general_suggestion_models

(would like to have called it suggestion_models, but that overlaps an existing usage)

Reply:
Actually, the main reason for "models" is the confusion that would result otherwise between domain objects and storage models. It's important that these be kept distinct, so leaving things as-is would result in a different type of confusion that we are trying to avoid throughout the codebase.

Reply:
Yeah I understand that differentiating between storage and domain objects is important. Is there any fix for when the imported model names conflict with the variable names for models that you create? Do you just add general_ or something as a prefix like you suggested here?

Reply:
I used general_ because that's the actual name of the model. But in general we don't have a convention for this yet. Perhaps we should...